core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/StructuredStreamingPlugin.scala

```diff
@@ -0,0 +1,109 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.harvester.plugin.embedded
+
+import org.apache.spark.sql.SaveMode
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.execution.streaming.FileStreamSink
+import org.apache.spark.sql.sources.DataSourceRegister
+import org.apache.spark.sql.sources.v2.writer.streaming.StreamWriter
+import org.apache.spark.sql.sources.v2.writer.{DataWriterFactory, WriterCommitMessage}
+import za.co.absa.commons.reflect.ReflectionUtils.extractFieldValue
+import za.co.absa.commons.reflect.extractors.SafeTypeMatchingExtractor
+import za.co.absa.spline.harvester.builder.SourceIdentifier
+import za.co.absa.spline.harvester.plugin.Plugin.{Precedence, ReadNodeInfo, WriteNodeInfo}
+import za.co.absa.spline.harvester.plugin.embedded.StructuredStreamingPlugin._
+import za.co.absa.spline.harvester.plugin._
+
+import javax.annotation.Priority
+
+@Priority(Precedence.Normal)
+class StructuredStreamingPlugin extends Plugin with ReadNodeProcessing with WriteNodeProcessing {
+
+  override val readNodeProcessor: PartialFunction[LogicalPlan, ReadNodeInfo] = {
+    case `_: StreamingDataSourceV2Relation`(relational) =>
+      val dataSourceReader = extractFieldValue[AnyRef](relational, "reader")
+      val options = extractFieldValue[Map[String, String]](relational, "options")
+
+      dataSourceReader match {
+        case `_: KafkaMicroBatchReader`(_) =>
+          val bootstrapServers = options("kafka.bootstrap.servers")
+          val topic = options("subscribe")
+          (SourceIdentifier(Some("kafka"), asURI(topic)), Map("bootstrap.servers" -> bootstrapServers))
+      }
+  }
+
+  override val writeNodeProcessor: PartialFunction[LogicalPlan, WriteNodeInfo] = {
+    case `_: WriteToDataSourceV2`(writeToDataSource) =>
+      val writer = extractFieldValue[AnyRef](writeToDataSource, "writer")
+      val query = extractFieldValue[LogicalPlan](writeToDataSource, "query")
+
+      writer match {
+        case `_: MicroBatchWriter`(microBatchWriter) =>
+          val dataSourceWriter = extractFieldValue[AnyRef](microBatchWriter, "writer")
+          dataSourceWriter match {
+
+            case `_: KafkaStreamWriter`(kafkaStreamWriter) =>
+              val topic = extractFieldValue[Option[String]](kafkaStreamWriter, "topic")
+              val producerParams = extractFieldValue[Map[String, String]](kafkaStreamWriter, "producerParams")
+              val bootstrapServers = producerParams("bootstrap.servers")
+
+              (SourceIdentifier(Some("kafka"), asURI(topic.get)), SaveMode.Append, query, Map("bootstrap.servers" -> bootstrapServers))
```

Comment:
Why not pass all producerParams?

Reply:
Yes, why not. Fixed

Reply:
ok

Reply:
As discussed, we will leave this as it is for now.